esp-hal/src/lib.rs

```diff
@@ -405,9 +424,40 @@ pub struct Blocking;
 ///
 /// Drivers are constructed in blocking mode by default. To set up an async
 /// driver, a [`Blocking`] driver must be converted to an `Async` driver using
-/// the `into_async` method. Drivers can be converted back to blocking mode
-/// using the `into_blocking` method.
+/// the `into_async` method, for example:
+///
+/// ```rust, no_run
+#[doc = crate::before_snippet!()]
+/// # use esp_hal::uart::{Config, Uart};
+/// let uart = Uart::new(
+///     peripherals.UART0,
+///     Config::default())?
+/// .with_rx(peripherals.GPIO1)
+/// .with_tx(peripherals.GPIO2)
+/// .into_async();
+///
+/// # Ok(())
+/// # }
+/// ```
+/// 
+/// Drivers can be converted back to blocking mode using the `into_blocking`
```

Comment:
I think part can just reference the `Blocking` struct again. Something like "see `Blocking` for more details"